pkg/asset/templates/kubelet.yaml

```diff
@@ -27,17 +27,11 @@ spec:
         - --api-servers={{ index .APIServers 0 }}
         - --config=/etc/kubernetes/manifests
         - --allow-privileged
-        - --hostname-override=$(MY_POD_IP)
```

Comment:
We should probably use the node name here. Then however the on-host kubelet has registered with the apiserver (IP or hostname), the self-hosted kubelet will always use the same.

There isn't currently support for the node name in the downward api., but there is an open PR: https://github.com/kubernetes/kubernetes/pull/27880